runtime/integration-tests/src/xcm/development/tests/connectors.rs

```diff
@@ -141,15 +141,137 @@ fn add_tranche() {
+
+		// Now create the pool
+		let pool_id: u64 = 42;
+		utils::create_pool(pool_id);
+
+		// Find the right tranche id
+		let pool_details = PoolSystem::pool(pool_id).expect("Pool should exist");
+		let tranche_id = pool_details
+			.tranches
+			.tranche_id(TrancheLoc::Index(0))
+			.expect("Tranche at index 0 exists");
+
+		Loans::update_nav(RuntimeOrigin::signed(ALICE.into()), pool_id.clone())
+			.expect("Should update nav");
+
+		// Finally, verify we can call Connectors::add_tranche successfully
+		// when given a valid pool + tranche id pair.
+		let new_member = DomainAddress::EVM(1284, [3; 20]);
+		let valid_until = 2555583502;
+
+		// Verify it fails if the origin is not a MemberListAdmin
+		assert_noop!(
+			Connectors::update_member(
+				RuntimeOrigin::signed(ALICE.into()),
+				new_member.clone(),
+				pool_id.clone(),
+				tranche_id.clone(),
+				valid_until.clone(),
+			),
+			BadOrigin
+		);
+
+		// Make ALICE the MembersListAdmin of this Pool
+		assert_ok!(Permissions::add(
+			RuntimeOrigin::root(),
+			Role::PoolRole(PoolRole::PoolAdmin),
+			ALICE.into(),
+			PermissionScope::Pool(pool_id.clone()),
+			Role::PoolRole(PoolRole::MemberListAdmin),
+		));
+
+		// Verify it now works
+		assert_ok!(Connectors::update_member(
+			RuntimeOrigin::signed(ALICE.into()),
+			new_member.clone(),
+			pool_id.clone(),
+			tranche_id.clone(),
+			valid_until.clone(),
+		));
+
+		// Verify the Investor role was set as expected in Permissions
+		assert!(Permissions::has(
+			PermissionScope::Pool(pool_id.clone()),
+			new_member.into_account_truncating(),
+			Role::PoolRole(PoolRole::TrancheInvestor(
+				tranche_id.clone(),
+				valid_until.clone()
+			)),
+		));
+
+		// Verify it can be called for another member
+		assert_ok!(Connectors::update_member(
+			RuntimeOrigin::signed(ALICE.into()),
+			DomainAddress::EVM(1284, [9; 20]),
+			pool_id.clone(),
+			tranche_id.clone(),
+			valid_until.clone(),
+		));
+	});
+}
+
+#[test]
+fn update_token_price() {
+	TestNet::reset();
+
+	Development::execute_with(|| {
+		utils::setup_pre_requirements();
+
+		// Now create the pool
+		let pool_id: u64 = 42;
+		utils::create_pool(pool_id);
+
+		// Find the right tranche id
+		let pool_details = PoolSystem::pool(pool_id).expect("Pool should exist");
+		let tranche_id = pool_details
+			.tranches
+			.tranche_id(TrancheLoc::Index(0))
+			.expect("Tranche at index 0 exists");
+
+		// Verify we first need to call `Loands::update_nav`
+		// Verify it fails if the origin is not a MemberListAdmin
+		assert_noop!(
+			Connectors::update_token_price(
+				RuntimeOrigin::signed(ALICE.into()),
+				pool_id.clone(),
+				tranche_id.clone(),
+				Domain::EVM(1284),
+			),
+			pallet_connectors::Error::<development_runtime::Runtime>::MissingTranchePrice
+		);
+
+		Loans::update_nav(RuntimeOrigin::signed(ALICE.into()), pool_id.clone())
+			.expect("Should update nav");
+
+		// Verify it now works
+		assert_ok!(Connectors::update_token_price(
```

Comment:
Question: Shouldn't we compare the token price before and after the invocation of `update_token_price` similar to the above permissions checks?

Reply:
This won't actually test the effect of having the connector message reach the smart contract, we are only testing here all the logic and requirements that need to be in place on the cent-chain to make these calls successfully.